include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -167,7 +183,7 @@ __pattern_transform_scan_base(_ExecutionPolicy&& __exec, _Iterator1 __first, _It
     const auto __n = __last - __first;
 
     // This is a temporary workaround for an in-place exclusive scan while the SYCL backend scan pattern is not fixed.
-    const bool bInplaceExclusiveScan = __n > 1 && !_Inclusive{} && __first == __result;
+    const bool bInplaceExclusiveScan = __n > 1 && !_Inclusive{} && __is_equal_iterators(__first, __result);
```

Comment:
We do not use the Hungarian notation, and we (sort of have to) use uglification. This variable is better named `__is_scan_inplace_exclusive`

Also please add all needed underscores to the `__is_equal_iterators` helper.

Reply:
Fixed.